### ISSUE TITLE
Refresh token functionality doesn't work properly

### ISSUE THREAD

**Issue by Ghostbird** [bcc-code/bmm-web#217]
Since #204 the app stops working at some point, and you have to manually refresh to fix it.

The good: It's cleaner than the old functionality, apparently the refresh token is used under the hood, as no browser page is needed.
The bad: There's no indication to the user what is wrong, and stuff just stops working, because under the hood every call returns 401.

We probably need to catch access token related 401s and force an access token refresh.

**Comment by SimonSimCity** [comment]
@Ghostbird What do you mean by "doesn't work properly"? What I've discovered so far is just #233 and a bug, that you have to reload after every successful login with redirect to the identity provider (seems to be something with vue's routing...) - this is fixed by manual refreshing. Is there anything else you know of?

**Comment by Ghostbird** [comment]
Exactly that. You need to manually refresh, though it should do so
automatically.

**Comment by SimonSimCity** [comment]
Well, the reason I see is that there's an error when redirecting. When you reload, it's not redirecting anymore, and the page loads just fine. You should see a bunch of error messages in the console and a white page - that's it in my case. If there are any other, please let me know.

**Comment by SimonSimCity** [comment]
I was able to nail it down to a quite small repo (https://stackblitz.com/edit/github-fn78bb) where I can reproduce this error. I'm currently waiting on the Vue team for this to be fixed: https://github.com/nuxt/nuxt/issues/13309

Another issue which seems to be linked also is https://github.com/vuejs/core/issues/6463

**Comment by kkuepper** [comment]
I applied the wontfix label, since (hopefully) nuxt needs to do the work and not us.

**Comment by kkuepper** [comment]
The nuxt and vue issues have been closed. Is this fixed? I don't think I've seen this problem recently.
If it's not fixed can you provide some reproduction steps?

**Comment by SimonSimCity** [comment]
Was fixed by updating vue to `v3.4.0`, which was already done in https://github.com/bcc-code/bmm-web/pull/269.
